File: php/advent-of-code/2025/shared.php
<?php 

  $BLUE = "\x1b[34m";
  $RED = "\x1b[31m";
  $RESET = "\x1b[0m";

  function plog ($msg, $type = "INFO") {
    $time = date("Y-m-d H:i:s");
    $COLOR = $type == "SUCCESS" ? "\x1b[34m" : "\x1b[31m";
    $RES = "$COLOR $time [$type]: $msg \x1b[0m\n";
    echo $RES;
    if ($type === "ERROR") {
      die($RES);
    }
  }

  function pSuccess($msg) {
    plog($msg, "SUCCESS");
  }

  function pErr($msg) {
    plog($msg, "ERROR");
  }

  
?>

File: lua/advent-of-code/2025/shared.lua
BLUE, RED, RESET = "\x1b[34m","\x1b[31m","\x1b[0m"

function now()
  return os.date("!%Y-%m-%d %H:%M:%S")
end

function perr(msg) 
  print(RED .. now() .. " LUA [Error]: " .. tostring(msg) .. RESET)
  error()
end

function plog(msg) 
  print(BLUE .. now() .. " LUA [RESULT]: " ..  " " .. tostring(msg) .. RESET)
end

function performance(main) 
  local perx = os.clock()
  if type(main) == 'function' then
    main()
  end
  print("elapsed time: ", os.clock() - perx)
end